Rewritten parser tests and a bugfix

MartinAltmayer opened the pull request:
This PR mainly contains rewritten tests for the parser package. Also, a bug found when writing the new tests was fixed: With the old code signatures would sometimes end up in Entity.Text and sometimes in Entity.Attachment, depending on the Content-Disposition header. This distinction has been replaced by a single field Entity.Content and a flag Entity.IsAttachment.

Commit:
Improved mail parser tests. Added some test mails.

Commit:
Fixed parsing of attachments.

Commit:
Added more parser tests.

Commit:
Added a comment.

Pharb commented on mail/parser.go:
```diff
@@ -174,14 +174,15 @@ func (parser *Parser) parseMultipartSigned(contentType *MimeMediaType, header te
-	signatureHeader := result.Parts[1].getHeader("Content-Type", "")
-	if signatureHeader != "application/pgp-signature" {
+	// Because we already successfully parsed the multipart content, no error can be returned here.
+	signatureHeader, _ := getMimeMediaTypeFromHeader(result.Parts[1].Header, "Content-Type", "")
```
"no error can be returned here" is a bold statement, I would prefer to check the returned error value and call something like `panic("should never happen")` if not `nil`.

Pharb commented on test/mails/plaintext.eml:
```diff
@@ -1,11 +1,10 @@
 Mime-Version: 1.0
-Content-Type: text/plain;
- charset=UTF-8
+Content-Type: text/plain
```
What is the reason for this change here? This is actually a stripped down version of an email sent by Github.

Pharb commented:
I'll fix the open issues and merge this, so I can use this for #9.

Pharb committed:
Fix linter, simplify redundant test code.

Pharb committed:
review comment: move "subject" header back to original position.

Pharb committed:
review comment: panic on "impossible" error.
Even if an error seems impossible in the current state, we should at least panic to avoid unexpected behaviour in the future.

Pharb commented:
Affects #6 & #9.
